gwcs/geometry.py

```diff
@@ -200,7 +201,13 @@ def evaluate(self, x, y, z):
         lon[h == 0] *= 0
 
         if self._wrap_lon_at != 180:
-            lon = np.mod(lon, 360.0 * u.deg if nquant else 360.0)
+
```

Comment:
I think this is more readable if instead of catching the warning the call is modified as in

`lon = np.mod(lon, 360.0 * u.deg if nquant else 360.0, where=np.isfinite(lon), out=lon)`